docs/maintainer/infrastructure.md

```diff
@@ -509,15 +509,172 @@ This was sponsored by Prefix.dev. Below are the relevant pull requests for the s
+
+```bash
+conda install compilers
+```
+
+Alternatively, specific compilers can be requested:
+
+```bash
+conda install cxx-compiler fortran-compiler
+```
+
+Other available convenience packages are listed in the [platform-default
+compilers](#platform-default-compilers) list.
+
+Please note that these packages must not be used in a feedstock; instead the macros
+listed in [Using compilers in feedstocks](#using-compilers-in-feedstocks) must be used.
+
+### MSVC compiler on Windows
+
+conda-forge is not allowed to redistribute the MSVC compiler used on Windows,
+and therefore only provides the activation scripts for an externally installed
+compiler. Users who wish to compile code using MSVC, both for local use
+and feedstock builds, have to install Microsoft Visual Studio manually.
+For more information, see [Notes on native code](/docs/maintainer/knowledge_base/#notes-on-native-code)
+in Knowledge Base.
+
+### Compilers supplied by conda-forge
+
+#### Platform-default compilers
+
+Currently conda-forge providers compilers for the following languages, that
+can be specified as arguments to the `{{ compiler(...) }}` macro:
+
+- `c`, also provided by `c-compiler` package
+- `cxx` (C++), also provided by `cxx-compiler`
+- `fortran`, also provided by `fortran-compiler`
+- `cuda`, also provided by `cuda-compiler`; see also [Guide for Maintainers of
+  Recipes That Use CUDA](https://github.com/conda-forge/cuda-feedstock/blob/main/recipe/doc/recipe_guide.md)
+- `rust`; see [Rust packages](/docs/maintainer/example_recipes/rust/)
+- `go-cgo` and `go-nocgo`; see [Go packages](/docs/maintainer/example_recipes/go/)
+
+The authoritative source of the current compilers and versions for various languages
+and platforms is the [conda_build_config.yaml](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml) file
+in the [conda-forge/conda-forge-pinning-feedstock](https://github.com/conda-forge/conda-forge-pinning-feedstock) repository
+as described in [Globally pinned packages](pinning_deps.md#globally-pinned-packages).
+
+The default C and C++ compilers are GCC on Linux, Clang on macOS and VS2022
+on Windows. Clang can also be used as the compiler on Linux and Windows.
+The default Fortran compiler is the GNU Fortran compiler on Linux and macOS,
+and Flang on Windows.
+
+Note that when used in conjunction with CUDA, the GCC version is restricted by the
+maximum version supported by nvcc (which is also reflected in the global pinning).
+
+#### clang vs. clang-cl on Windows
+
+The `clang` compiler package installs two frontends, and conda-forge
+provides separate activation scripts for Windows, for each of them. Therefore,
+the following arguments can be used in `recipe/conda_build_config.yaml`
+or passed to the `{{ compiler(...) }}` macro:
+
+- `clang` to use the `clang` frontend using GCC argument syntax
+- `clang-cl` to use the `clang-cl` frontend with MSVC argument syntax
```

Comment:
OK, I think this is based on a misunderstanding of my comment. I meant that one can do either:
```yaml
c_compiler:  # [win]
  - clang    # [win]
```
OR
```yaml
c_compiler:   # [win]
  - clang-cl  # [win]
```